src/reqwest.rs

```diff
@@ -4,39 +4,23 @@ use thiserror::Error;
 /// Error type returned by failed reqwest HTTP requests.
 ///
 #[derive(Debug, Error)]
-pub enum Error<T>
-where
-    T: std::error::Error + 'static,
-{
+pub enum Error {
     /// Error returned by reqwest crate.
     #[error("request failed")]
-    Reqwest(#[source] T),
+    Reqwest(#[from] reqwest::Error),
     /// Non-reqwest HTTP error.
     #[error("HTTP error")]
-    Http(#[source] http::Error),
+    Http(#[from] http::Error),
     /// I/O error.
     #[error("I/O error")]
-    Io(#[source] std::io::Error),
-    /// Other error.
-    #[error("Other error: {}", _0)]
```

Comment:
If we're going to remove this variant, we should probably make the enum `#[non_exhaustive]` so we'll still have an escape hatch to avoid future breaking changes. In any case, we'll need `Other` or another variant for errors that arise during the `url::Url` to `http::Uri` conversion.

Reply:
Sounds good; let's go for `non_exhaustive` to have an escape-hatch until we define what the error type is going to be for that conversion.